core/src/test/java/org/testcontainers/containers/GenericContainerTest.java

```diff
@@ -60,6 +72,53 @@ public void shouldReportErrorAfterWait() {
         }
     }
 
+    @Test
+    public void shouldOnlyPublishExposedPorts() {
+        ImageFromDockerfile image = new ImageFromDockerfile("publish-multiple")
+            .withDockerfileFromBuilder(builder ->
+                builder
+                    .from("testcontainers/helloworld:1.1.0")
+                    .expose(8080, 8081)
+                    .build()
+            );
+        try (GenericContainer<?> container = new GenericContainer<>(image).withExposedPorts(8080)) {
+            container.start();
+
+            InspectContainerResponse inspectedContainer = container.getContainerInfo();
+
+            List<Integer> exposedPorts = Arrays.stream(inspectedContainer.getConfig().getExposedPorts())
+                .map(ExposedPort::getPort)
+                .collect(Collectors.toList());
+
+            assertEquals(
+                "the exposed ports are all of those EXPOSEd by the image",
+                Ints.asList(8080, 8081),
+                exposedPorts
+            );
+
+            Map<ExposedPort, Ports.Binding[]> hostBindings = inspectedContainer.getHostConfig().getPortBindings().getBindings();
+            assertEquals(
+                "only 1 port is bound on the host (published)",
+                1,
+                hostBindings.size()
+            );
+
+            Integer mappedPort = container.getMappedPort(8080);
+            assertTrue(
+                "port 8080 is bound to an ephemeral port on the host",
+                mappedPort >= 32768
```

Comment:
nit: I think this depends on environment configuration. Can we just check that `mappedPort != 8080`?